include/DoocsBackendNumericRegisterAccessor.h

```diff
@@ -0,0 +1,371 @@
+    }
+  }
+
+  /********************************************************************************************************************/
+
+  template<typename UserType>
+  template<typename T>
+  T DoocsBackendNumericRegisterAccessor<UserType>::dataGet(int index) {
+    switch(this->dst.type()) {
+      case DATA_BOOL:
+        return this->dst.get_bool();
+
+      case DATA_A_BOOL:
+      case DATA_SHORT:
+      case DATA_A_SHORT:
+        return this->dst.get_short(index);
+
+      case DATA_USHORT:
+      case DATA_A_USHORT:
+        return this->dst.get_ushort(index);
+
+      case DATA_INT:
+      case DATA_A_INT:
+      case DATA_IIII:
+        return this->dst.get_int(index);
+
+      case DATA_UINT:
+      case DATA_A_UINT:
+        return this->dst.get_uint(index);
+
+      case DATA_LONG:
+      case DATA_A_LONG:
+        return this->dst.get_long(index);
+
+      case DATA_ULONG:
+      case DATA_A_ULONG:
+        return this->dst.get_ulong(index);
+
+      case DATA_IFFF:
+        if(index == 0) {
+          return this->dst.get_int(index);
+        }
+        return this->dst.get_float(index);
+
+      case DATA_FLOAT:
+      case DATA_SPECTRUM:
+      case DATA_GSPECTRUM:
+      case DATA_A_FLOAT:
+        return this->dst.get_float(index);
+
+      case DATA_DOUBLE:
+      case DATA_A_DOUBLE:
+      default:
+        return this->dst.get_double(index);
+    }
+  }
+
+  /********************************************************************************************************************/
+
+  template<>
+  inline void DoocsBackendNumericRegisterAccessor<ChimeraTK::Void>::doPostRead(TransferType type, bool hasNewData) {
+    DoocsBackendRegisterAccessor<ChimeraTK::Void>::doPostRead(type, hasNewData);
+  }
+
+  /********************************************************************************************************************/
+
+  template<typename UserType>
+  void DoocsBackendNumericRegisterAccessor<UserType>::doPostRead(TransferType type, bool hasNewData) {
+    DoocsBackendRegisterAccessor<UserType>::doPostRead(type, hasNewData);
+    if(!hasNewData) return;
+
+    // special workaround for D_spectrum: Data type will be DATA_NULL if error is set to "stale data"
+    if(this->dst.type() == DATA_NULL) {
+      // Unfortunately there is no way to get to the data at this point, so fill the buffer with zeros. The data
+      // validity has been set to invalid in this case already by the
+      // DoocsBackendRegisterAccessor<UserType>::doPostRead() call above.
+      return;
+    }
+
+    // verify array length
+    if(size_t(this->dst.length()) < this->nElements + this->elementOffset) {
+      throw ChimeraTK::runtime_error("DoocsBackend: Unexpected array length found in remote property " +
+          this->ea.show_adr() + ": " + std::to_string(this->dst.length()) + " is shorter than the expected " +
+          std::to_string(this->nElements));
```

Comment:
The error message can be wrong if the offset is not zero. For instance dst.length() is 10 and nElements is 5 and offset is 8, the message reads "10 is shorter than 5", which is not true. It should read "10 is shorter than 5 + 8 (nElements + offset)"

Reply:
fixed